src/operations/execute_operation.ts

```diff
@@ -246,24 +235,17 @@ function executeWithServerSelection(
   }
 
   // select a server, and execute the operation against it
-  topology.selectServer(selector, serverSelectionOptions, (err?: any, server?: any) => {
-    if (err) {
-      callback(err);
-      return;
+  topology.selectServer(selector, serverSelectionOptions, (error, server) => {
+    if (error || !server) {
+      return callback(error);
     }
 
     if (session && operation.hasAspect(Aspect.RETRYABLE)) {
-      const willRetryRead =
-        topology.s.options.retryReads !== false &&
-        !inTransaction &&
-        supportsRetryableReads(server) &&
-        operation.canRetryRead;
-
-      const willRetryWrite =
-        topology.s.options.retryWrites === true &&
-        !inTransaction &&
-        supportsRetryableWrites(server) &&
-        operation.canRetryWrite;
+      const willRetryRead = topology.s.options.retryReads;
```

Comment:
this and the willRetryWrite blocks should not have semicolons

Reply:
Fixed!